api/rds.proto

```diff
@@ -137,6 +137,16 @@ message RouteAction {
     WeightedCluster weighted_clusters = 3;
   }
 
+  enum ClusterNotFoundResponseCode {
+    // HTTP status code - 503 Service Unavailable.
+    SERVICE_UNAVAILABLE = 0;
+    // HTTP status code - 404 Not Found.
+    NOT_FOUND = 1;
+  }
+  // The HTTP status code to use when configured cluster is not found.
```

Comment:
Are referenced clusters warmed? What happens if the cluster is found but EDS has not completed?

Reply:
https://github.com/envoyproxy/envoy/pull/1971 covers only the case when cluster is configured (or provided via `cluster_header`), but cluster manager cannot find it.

I suspect that cold clusters are still visible to the cluster manager, but without endpoints you'd get `503 Service Unavailable`, but I'm just guessing here, really. cc @mattklein123

Reply:
cluster exists but no hosts will always be 503. I think that is probably fine to continue to be always the case.